src/_zkapauthorizer/tests/test_matchers.py

```diff
@@ -142,3 +164,274 @@ def test_matches(self):
+                Not(equals_database(other)),
+            ),
+        )
+
+    @given(
+        sql_schemas(dict_kwargs={"min_size": 1}).flatmap(
+            lambda schema: tuples(
+                # Pass along the schema so the test can create it.
+                just(schema),
+                # Build some arbitrary amount of data that fits into the
+                # schema so there are rows of data involved.
+                fixed_dictionaries(
+                    {
+                        name: lists(inserts(name, table))
+                        for (name, table) in schema.items()
+                    }
+                ),
+                # Create one more row of data which will be used to make the
+                # databases differ.
+                sampled_from(sorted(schema.items())).flatmap(
+                    lambda item: inserts(*item),
+                ),
+            ),
+        )
+    )
+    def test_different_rows(self, schema_and_common_and_different):
+        """
+        Two databases with the same schema but different rows in their tables do
+        not match.
+        """
+        schema, common_inserts, different_insert = schema_and_common_and_different
+        for name, table in schema.items():
+            sql = create_table(name, table)
+            self.original.execute(sql)
+
+        for name, statements in common_inserts.items():
+            for stmt in statements:
+                self.original.execute(stmt.statement(), stmt.arguments())
+
+        copied = copy(self.original)
+        copied.execute(different_insert.statement(), different_insert.arguments())
+
+        self.assertThat(
+            self.original,
+            Not(equals_database(copied)),
+        )
+
+    @given(
+        sql_schemas(dict_kwargs={"min_size": 1}).flatmap(
+            lambda schema: tuples(
+                # Pass along the schema so the test can create it.
+                just(schema),
+                # Build some arbitrary amount of data that fits into the
+                # schema so there are rows of data involved.
+                fixed_dictionaries(
+                    {
+                        name: lists(inserts(name, table))
+                        for (name, table) in schema.items()
+                    }
+                ),
+            ),
+        )
+    )
+    # Add some known problematic cases.  Hypothesis found these originally but
+    # let's help it keep an eye on them in the future, too.
+    @_float_example(
+        [
+            # Some examples our CI found
+            1.311946107307449e-10,
+            1.1466443538665771e-05,
+            1.500589370727539,
+            1125899906842624.0,
+            5.192298096474867e33,
+            # This one has a distance of 3
+            -1.1919735848895067e-35,
+        ]
+    )
+    @_float_example(
+        [
+            # Examples from Rick Regan's blog post that are broken on Windows.
```

Comment:
Maybe add the URL since you don't have his name above where you have the URL:

https://www.exploringbinary.com/incorrect-decimal-to-floating-point-conversion-in-sqlite/